pkg/sdn/plugin/cniserver/cniserver.go

```diff
@@ -141,7 +143,11 @@ func (s *CNIServer) Start(requestFunc cniRequestFunc) error {
 	}
 
 	s.SetKeepAlivesEnabled(false)
-	go s.Serve(l)
+	go utilwait.Forever(func() {
+		if err := s.Serve(l); err != nil {
+			glog.Warningf("CNI server Serve() failed: %v", err)
```

Comment:
based on other sdn code, I think the preferred behavior is to call (kubernetes/pkg/util/)runtime.HandleError()